src/advanced_inv.cpp

```diff
@@ -1746,16 +1793,16 @@ bool advanced_inventory::add_item( aim_location destarea, const item &new_item,
         g->u.moves -= 100;
         return rc;
     } else if(destarea == AIM_WORN) {
-        // add and equip item, note this can duplicate if proper removal is not done post call.
-        if(inv_item != nullptr) {
-            // return the item's pointer, for removal after call is done
-            *inv_item = &g->u.i_add(new_item);
-            return g->u.wear_item(*inv_item);
-        } else {
-            return g->u.wear_item(&g->u.i_add(new_item));
+        if((rc = g->u.wear_item(&new_item)) == false) {
+            popup(_("You can't wear that!"));
         }
+        return rc;
     } else {
         advanced_inv_area &p = squares[destarea];
+        // set the vehicle area for the destination area
+        if(squares[destarea].can_store_in_vehicle()) {
```

Comment:
`can_store_in_vehicle` returns true if there is a vehicle there, it does not check whether I actually *want* to put it into the vehicle. Selecting a square that has a vehicle (but not selecting the vehicle storage) will make this condition true. Consequently `set_vehicle` is called, which calls `swap_panes` and causes the "This here be a bug! [NUM_AIM_LOCATIONS] in set_vehicle()" debug message (twice, a second time from `remove_item` for the same reason).

I think you might be better of with a simply bool in `advanced_inventory_pane` and in `advanced_inventory_area` that indicates "use the vehicle storage (if it exists)" instead AIM_VEHICLE, and consequently remove "AIM_VEHICLE" completely.